infrastructure/kctf-samples/apache-php/challenge/config/nsjail-php.cfg

```diff
@@ -4,8 +4,8 @@ description: "PHP CGI nsjail configuration for web RCE CTF task."
 mode: ONCE
 chroot_dir: "/chroot"
 log_level: ERROR
-uidmap {inside_id: "1000"}
-gidmap {inside_id: "1000"}
+# Run as nobody so session files can be saved in GCS
```

Comment:
this is ugly, but I can't find a way to map the outside id 1000 to the inside id 1000 when running as www-data unless SUID. this is a bit of a hack, but works.

Reply:
I don't think that runs as nobody, it probably runs as the current user instead? Can you make the web server use uid 1000 instead of www-data?

Reply:
it runs as the current user outside, but inside its mapped to the uid of nobody.

I tried running apache as uid 1000 but that also had issues when mapping, although I didn't try too hard once I found this worked. I'll try again

Reply:
or alternatively pre-create the www-data user both inside and outside and do an identity mapping

Reply:
all files in /mnt/drives/gcs-task are owned by user 1000

Reply:
now apache runs as 1000